lib/generator.ml

```diff
@@ -2,23 +2,54 @@ open Json_schema_t
 open Printf
 open Utils
 
-let record_field_name str =
+type state = {
+  with_doc : bool;
+  protect_against_duplicates : string list ref option;
+  toplevel_types : [ `All | `Only of string list ];
+  avoid_dangling_refs : bool;
+  json_ocaml_type : [ `From | `Module ] * string * string;
+}
+
+let default_state =
+  {
+    with_doc = true;
```

Comment:
Not sure it it should be a default value, or at least tests should be adjusted accordingly